lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -43,14 +43,51 @@
  *
  */
 
+#include <functional>
 #include "lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.h"
 #include "aten/src/ATen/WrapDimUtils.h"
 #include "torch/csrc/api/include/torch/enum.h"
-
 namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, bool unbiased){
+  return compute_shape_std(self, c10::nullopt, c10::nullopt, false);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, at::IntArrayRef dim, bool unbiased, bool keepdim){
+  return compute_shape_std(self, dim, c10::nullopt, keepdim);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, c10::optional<at::IntArrayRef> dim, c10::optional<int64_t> correction, bool keepdim){
+  if (dim.has_value()) {
+    auto sizes = self.sizes().vec();
+    if (keepdim) {
+      for (auto reduce_dim: dim.value()) {
+        sizes[at::maybe_wrap_dim(reduce_dim, self)] = 1;
+      }
+    } else {
+      std::vector<int64_t> reduce_dims(dim.value().begin(), dim.value().end());
+      at::maybe_wrap_dims(reduce_dims, self.dim());
+      sort(reduce_dims.begin(), reduce_dims.end(), std::greater<int64_t>());
```

Comment:
Where does this sorting logic come from? I'm staring at the code for `make_reduction()`, and the case where `keepdim=False` seems pretty similar to the `true` case ([here](https://github.com/pytorch/pytorch/blob/0bd0a67c4fce0ff9f005ae38d10df5ac778e7a54/aten/src/ATen/native/ReduceOpsUtils.h#L134)). Something like:
```
  auto shape = DimVector(self.sizes());
  for (int dim = shape.size() - 1; dim >= 0; dim--) {
    if (mask[dim]) {
      if (keepdim) {
        shape[dim] = 1;
      } else {
        shape.erase(shape.begin() + dim);
      }
    }
  }
  return shape;
```

Reply:
ahh. i should have used that code, i'll change to use it. 

In their version they construct a 'dimmask' which is equivalent to my sorting, in terms of canonicalizing the keepdims in some order, so that you can safely remove them from the vector starting from the higher indices toward the lower indicies.

The issue would be if you tried to remove dim 1 then later remove dim 3, you'd have to keep track of the indices above the thing you remove all shifting down.